development/extensions/skeleton_extension.rst

```diff
@@ -0,0 +1,750 @@
+
+Author role
+^^^^^^^^^^^
+
+The role of an extension's author. For example, lead developer, contributor, 
+translator, etc. (Optional)
+
+Requirements
+------------
+
+Requirements are where you define the minimum (and maximum) PHP and phpBB versions
+your extension needs to be able to perform.
+
+Minimum phpBB requirement of the extension
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Typically this should be set to the version of phpBB in which you are developing
+and testing your extension.
+
+The version should be defined using `Composer's version 
+constraints <https://getcomposer.org/doc/articles/versions.md>`_. 
+
+For example, to require phpBB 3.1.4 or greater, use ``>=3.1.4``.
+
+Maximum phpBB requirement of the extension
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To mitigate compatibility breaks, it is common to set the maximum version
+to the next major release version of phpBB.
+
+For example, if your extension is being developed for phpBB 3.1.x, you can
+set the maximum requirement to phpBB 3.2.x by using ``<3.2.0@dev``.
+
+PHP requirement of the extension
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Since your extension will operate within phpBB, it's often safe to assume
+that your extension will share the same minimum PHP requirement as phpBB
+does. Thus, by default, this field will use ``>=5.3.3`` unless you enter
+an alternative version.
+
+.. note::
+
+    You can find phpBB's PHP requirements by looking at its composer.json file.
+
+Components
+----------
+
+The default Skeleton Extension outputs a license file and composer.json.
+The Components section allows you to choose from a variety of typical
+components an extension could need, such as an ACP module, migrations,
+events, and tasks such as cron, CLI and notifications. The files
+generated for these components are simple yet fully working examples
+based off the Acme Demo extension. They are intended to give you a
+consistent and phpBB code-ready starting point, so you can focus more
+on writing your code and less on file structure and organization.
+
+PHP event listeners
+-------------------
+
+PHP event listeners work with core events to inject code into phpBB.
+Core events are like hooks, and they can be found throughout phpBB's
+codebase at key points. They give your extension access to phpBB's
+variables and allow you to use and modify them or to inject additional
+PHP code during phpBB's execution.
+
+The Skeleton Extension will generate a sample event listener PHP file
+and necessary services YAML file:
+
+::
+
+    vendor
+    ├── package
+    │   ├── config                 # The config dir contains all service config files
+    │   │   ├── services.yml       # A config YAML file
+    │   │   └── ...
+    │   ├── event                  # The event dir contains all PHP event listeners
+    │   │   ├── main_listener.php  # A sample PHP event listener
+    │   │   └── ...
+    │   └── ...
+    └── ...
+
+Styles listeners
```

Comment:
Not sure what you mean. In the ext skeleton this is called Style listeners, and we explain these are template events and display the template events the skeleton generates.